Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,352 @@
+    {
+        $this->declareChannel();
+        $this->declareQueue($message->getQueue(), AMQP_DURABLE, $arguments);
+        $messageBody = $this->getSerializer()->serialize($message, $this->format);
+        $messageHeaders = new AMQPTable($message->getHeaders());
+        $amqpMessage = new AMQPMessage($messageBody);
+        $amqpMessage->set('application_headers', $messageHeaders);
+        $this->channel->basic_publish($amqpMessage, self::EXCHANGE_NAME, $message->getQueue());
+        return true;
+    }
+
+    /**
+     * Declares the queue to drive the message.
+     *
+     * @param string $queueName The name of the queue
+     * @param int $durable
+     * @param array $arguments See AMQPQueue::setArguments()
+     *
+     * @return array|null
+     */
+    public function declareQueue(string $queueName, int $durable = AMQP_DURABLE, array $arguments = [])
+    {
+        return $this->channel->queue_declare($queueName, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Catch a channel inside the connection.
+     *
+     * @param int $prefetchSize
+     * @param int $prefetchCount
+     * @return AMQPChannel
+     * @throws \Exception
+     */
+    public function declareChannel(int $prefetchSize = self::PREFETCH_SIZE, int $prefetchCount = self::PREFETCH_COUNT): AMQPChannel
+    {
+        $this->validateConnection();
+        $this->channel = $this->amqpConnections[0]->channel();
+        $this->channel->basic_qos($prefetchSize, $prefetchCount, null);
+
+        return $this->channel;
+    }
+
+    /**
+     * Returns the format used on serialize function.
+     *
+     * @return string
+     */
+    public function getFormat(): string
+    {
+        return $this->format;
+    }
+
+    /**
+     * Set the format used by serialize/deserialize functions
+     *
+     * @param string $format
+     */
+    public function setFormat(string $format = null)
+    {
+        $this->format = $format;
+    }
+
+    /**
+     * Shuffles the connection in case there is more than one available.
+     */
+    protected function shuffleConnections()
+    {
+        shuffle($this->connectionsData);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function waitNoBlock()
+    {
+        $this->channel->wait(null, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function wait()
+    {
+        $this->channel->wait();
+    }
+
+    public function isConsuming(): bool
+    {
+        return $this->channel->is_consuming();
+    }
+
+    /**
+     * Set the port to connection
+     *
+     * @param int $port
+     */
+    public function setPort(int $port)
+    {
+        $this->port = $port;
+    }
+
+    /**
+     * Return one or more connections available on the driver
+     *
+     * @return array
+     */
+    public function getAvailableConnections(): array
+    {
+        return $this->amqpConnections;
+    }
```

Comment:
Is there a need to have a method to get all the amqpConnections?

Reply:
Not relevant anymore